core/src/replay_stage.rs

```diff
@@ -652,8 +653,14 @@ impl ReplayStage {
             let node_keypair = cluster_info.read().unwrap().keypair.clone();
 
             // Send our last few votes along with the new one
+            let mut last_vote = tower.last_vote();
+            if tower.needs_timestamp(bank.slot()) {
+                let timestamp = Utc::now().timestamp();
+                last_vote.timestamp = Some(timestamp);
+                tower.record_recent_timestamp(bank.slot(), timestamp);
```

Comment:
nice, you could combine the needs() and record() into a single function that returns an Option that you'd stuff directly into last_vote...  you could also have last_vote() take a slot and do all the work...

Reply:
Actually, if I do it that way, I don't even need the Slot, as I can take it from the last_vote. Rad.
I'm going to add a separate method, though, since `last_vote()` is used to print out the last vote a few places, and I don't want it updating the timestamp in those cases.